packages/react-sdk/src/components/link.tsx

```diff
@@ -1,13 +1,56 @@
+import { Link as RemixLink, useLocation } from "@remix-run/react";
 import { forwardRef, type ElementRef, type ComponentProps } from "react";
 
+const isAbsoluteUrl = (href: string) => {
+  // http:, mailto:, tel:, etc.
+  return /^[a-z]+:/i.test(href);
+};
+
+// @todo: Ideally this code should live in designer app.
+// Sdk should not know how routing is implemented in designer.
+// We can move it there later when we have a way of customizing components per environment.
+const preserveBuildParams = (href: string, currentSearch: string) => {
+  if (isAbsoluteUrl(href)) {
+    return href;
+  }
+
+  const url = new URL(href, "http://localhost");
+
+  const currentSearchParams = new URLSearchParams(currentSearch);
+
+  const mode = currentSearchParams.get("mode");
+  if (mode === "preview" || mode === "edit" || mode === "published") {
```

Comment:
we need to have a single source of truth for those modes since they are used in multiple places now and if we need to rename or add, it would be much simpler to figure out